src/autopas/options/Option.h

```diff
@@ -117,12 +134,14 @@ class Option {
       haystack.push_back(optionString);
     }
 
-    for (auto &needle : needles) {
-      // first find the best matching string
-      auto matchingString = autopas::utils::StringUtils::matchStrings(haystack, needle);
-      // then find the corresponding enum and add it to the return set
-      foundOptions.insert(allOptionNamesLower[matchingString]);
-    }
+    // convert all needles to options.
+    std::transform(needles.begin(), needles.end(), std::insert_iterator(foundOptions, foundOptions.begin()),
+                   [&](const auto &needle) {
+                     // first find the best matching string
+                     auto matchingString = autopas::utils::StringUtils::matchStrings(haystack, needle);
+                     // then find the corresponding enum and add it to the return set
+                     return allOptionNamesLower[matchingString];
+                   });
```

Comment:
Yes! The problem with the previous version was that it used `insert()`. This function has a different signature for `std::vector` and `std::set`. Since this function now should support any container type it was impossible to use `insert()`. Rewriting this part with `std::insert_iterator` provides enough abstraction to solve this.